### PR TITLE
Add new keyword "Wait until salesforce is ready"

### PR DESCRIPTION
**Pull request by boakley**
Work item [W-11241613](https://gus.lightning.force.com/a07EE00000yQFvbYAG)

This adds a new playwright keyword `Wait until salesforce is ready`, and calls it from `Open Test Browser`

This uses `Wait until loading is complete` but also adds an additional check for landing on a classic page.

This also adds a new 'wait' parameter to the playwright version of `Open test browser`, for parity with the selenium version. Plus, it makes testing this particular feature easier.

The wait times are admittedly a bit arbitrary, but they seem to work. If I reduce them much, they will fail more often when Salesforce is running slower than usual.

I don't recall if I'm still supposed to check in the documentation or if we regenerate it when we regenerate the rest of the docs, so I'm including it here.

# Changes

* There is a new playright-based keyword `Wait until salesforce is ready`. This keyword is automatically called by the `Open Test Browser` keyword in the SalesforcePlaywright library. In addition to waiting for the page to be rendered, it will auto attempt to detect a classic page on initial render, and automatically switch to a lightning page if one is detected.

### CODE REVIEW

File: cumulusci/robotframework/SalesforcePlaywright.py [hunk 1]
```diff
@@ -126,14 +133,83 @@ class 'slds-template__container', but a different locator can
...
+        keyword will actually wait for half of the timeout before
+        starting checks for edge cases.
+
+        """
+
+        timeout_seconds = timestr_to_secs(timeout)
+        start_time = time.time()
+
+        locator = locator if locator else "div.slds-template__container"
+        expected_url = rf"/{self.cumulusci.org.lightning_base_url}\/lightning\/.*/"
```
Comment (TheBitShepherd) [inline cumulusci/robotframework/SalesforcePlaywright.py:168]: ooo, the rare and ever elusive `raw format string` in the wild!

File: cumulusci/robotframework/SalesforcePlaywright.py [hunk 2]
```diff
@@ -126,14 +133,83 @@ class 'slds-template__container', but a different locator can
...
+        The timeout parameter is taken as a rough guideline. This
+        keyword will actually wait for half of the timeout before
+        starting checks for edge cases.
+
+        """
+
+        timeout_seconds = timestr_to_secs(timeout)
+        start_time = time.time()
+
+        locator = locator if locator else "div.slds-template__container"
```
Comment (TheBitShepherd) [inline cumulusci/robotframework/SalesforcePlaywright.py:167]: Could also use the `or` here to keep things even more concise
```suggestion
        locator = locator or "div.slds-template__container"
```

**Review by TheBitShepherd — APPROVED** [review]
Nice job @boakley!

File: cumulusci/robotframework/SalesforcePlaywright.py (hunk 1 above)
Comment (boakley) [reply cumulusci/robotframework/SalesforcePlaywright.py:168]: I think this is the first time I've ever used it.

**Pull request merged**